package/src/error.rs

```diff
@@ -0,0 +1,188 @@
+use std::path::{Path, PathBuf};
+
+use gix::ObjectId;
+use nickel_lang_core::{files::Files, identifier::Ident};
+
+/// Errors related to package management.
+pub enum Error {
+    Io {
+        path: Option<PathBuf>,
+        error: std::io::Error,
+    },
+    LockFileDeserialization {
+        path: PathBuf,
+        error: serde_json::Error,
+    },
+    ManifestEval {
+        package: Option<Ident>,
+        files: Files,
+        error: nickel_lang_core::error::Error,
+    },
+    NoPackageRoot {
+        path: PathBuf,
+    },
+    NoProjectDir,
+    RestrictedPath {
+        /// The url of the git package that tried the bad import.
+        package_url: Box<gix::Url>,
+        /// The git id of the bad package.
+        package_commit: ObjectId,
+        /// The relative path of the bad package within its git repo.
+        package_path: PathBuf,
+        attempted: PathBuf,
+        restriction: PathBuf,
+    },
+    /// There was some error interacting with a git repository.
+    Git(nickel_lang_git::Error),
+    InvalidUrl {
+        url: String,
+        msg: String,
+    },
+    Resolution {
+        msg: String,
+    },
+    InternalManifestError {
+        path: PathBuf,
+        msg: String,
+    },
+    /// There was an error persisting a temporary file.
+    TempFilePersist {
+        error: tempfile::PersistError,
+    },
+    NoManifestParent,
+    /// Index dependencies aren't implemented yet, so we emit
+    /// this if we encounter one.
+    IndexDep,
+}
+
+impl std::error::Error for Error {}
+
+impl std::fmt::Debug for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        std::fmt::Display::fmt(self, f)
+    }
+}
+
+impl std::fmt::Display for Error {
```

Comment:
I haven't tried, but I suppose this means that package management errors are displayed in a way that doesn't follow the other errors look-and-feel? It doesn't have to be done now, but would implementing `IntoDiagnostics` instead and re-use the Nickel error reporting infrastructure work as well?

Reply:
I don't have a strong opinion about this, but I did it this way because most of the package errors don't use the reporting machinery. In `cli/src/error.rs` you'll see that the `Display` implementation is used in conjunction with `report_with_msg` for most variants, and `ManifestEval` gets special handling. So the errors should have the same look-and-feel as the others.

Reply:
Ah, if it gets reported with the usual machinery, it's fine then. No need to import that here if you don't use the notes, labels or the like.